### PR TITLE
UI for identity server client

### PR DESCRIPTION
**Pull request by siddharthashw**
Fixes - #219 

**1. ConsumerAppController.cs** - update all the API and implement IdentityServer4 Client instead of ConsumerApp module, updated API doc
**2. OAuthController.cs** - deleted
**3. Data Folder** - Added migration and remove ConsumerApp module
**4. DataRepository.cs** - Added new method Any and updated the repository to generic of Database
**5. IDataRepository.cs** - added for generic database
**6. AllowedScope.cs** - added enum
**7. ConsumerApps.cs** - removed unnecessary property and added 3 new property
**8. ConsumerAppRepository.cs** - updated the whole repository to implement IdentityServer4 client and updated xml comment
**9. IConsumerAppRepository.cs** - updated the whole repository to implement IdentityServer4 client and updated xml comment
**10. OAuthRepository** - deleted
**11. ProjectRepository.cs** - used DataRepository with particular DB
**12. UserRepository.cs** - used DataRepository with particular DB
**13. IdentityServerInitialize.cs** - updated the seed method and used ConfigurationDbContext
**14. consumerapp-add.component.ts** - updated component to implement IdentityServer4 Client field and added functionality of generating AuthId and secret at front view
**15. consumerapp-add-html** - remove description field and added scope and logout field and added functionality of generating AuthId and secret at front view
**16. consumer-edit-component.ts** - updated component to implement IdentityServer4 Client field and added functionality of generating AuthId and secret at front view
**17. consumerapp-edit-html** - remove description field and added scope and logout field and added functionality of generating AuthId and secret at front view
**18. consumerapp-list.html** - change on click return field from id to clientId
**19. consumerapp-model.ts** - added enum consumerappallowedscopes and added 3 new property in ConsumerAppModel
**20. consumerapp.service.ts** - update the put method url
**21. project.json** - update the version to 1.0.0-preview4-final of -> Microsoft.AspNetCore.Razor.Tools, Microsoft.EntityFrameworkCore.Tools, Microsoft.VisualStudio.Web.CodeGeneration.Tools and Microsoft.VisualStudio.Web.CodeGenerators.Mvc
**22. Startup.cs** - remove DI of OAuthRepository
**23. consumer.service.ts** - added new service for generating random number
**24. http.service.ts** - added a get method which not expect object in return
**25. IStringConstant.cs** - added variables for test cases -> RandomClientId, RandomClientSecret, ExceptionMessageConsumerAppNameIsAlreadyExists and ExceptionMessageConsumerAppNotFound
**26. StringConstant.cs** - added variables for test cases -> RandomClientId, RandomClientSecret, ExceptionMessageConsumerAppNameIsAlreadyExists and ExceptionMessageConsumerAppNotFound
**27. BaseProvider.cs** - register IdentityServer4 dependencies
**28. ConsumerAppRepositoryTest.cs** - updated the all test cases
**29. project.json (Test)** - added package :-
"IdentityServer4.EntityFramework": "1.0.0",
"IdentityServer4": "1.0.2",
"IdentityServer4.AspNetIdentity": "1.0.0",
"IdentityServer4.AccessTokenValidation": "1.0.2"
**30. consumerapp-add-component-spec.ts** - updated test case and added two more test cases
**31. consumerapp-edit-component-spec.ts** - updated test case and added two more test cases
**32. mock.consumerapp.service.ts** - added mocking of getRandomNumber

**After accepting this pull request, please drop the db and restart the application.**

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs [hunk 1]
```diff
@@ -11,14 +8,15 @@
...
 {
-    [Authorize(Roles = "Admin")]
-    [Route("api/[controller]")]
+    [Authorize(Roles = Admin)]
+    [Route(BaseUrl)]
     public class ConsumerAppController : Controller
     {
         #region "Private Variable(s)"
         private readonly IConsumerAppRepository _consumerAppRepository;
         private readonly UserManager<ApplicationUser> _userManager;
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs]: Unused variable

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs [hunk 2]
```diff
@@ -172,23 +207,34 @@ public ConsumerAppController(IConsumerAppRepository iConsumerAppRepository, User
...
+        *       "isAuthId" : "true"
+        *  }
+        * @apiSuccessExample {json} Success-Response:
+        * HTTP/1.1 200 OK 
+        * {
+        *       "DADEGSFGFDVGD"
+        * }
+        */
+        [HttpGet]
+        [Route("generaterandomnumber/{isAuthId:bool}")]
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs]: wrong API url

Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs]: What is the use of RandomNumber Generator here?

Comment (siddharthashw) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ConsumerAppController.cs]: For generating random string for AuthId and AuthSecret

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ConsumerAppRepository/ConsumerAppRepository.cs [hunk 3]
```diff
@@ -68,69 +79,210 @@ public ConsumerAppRepository(IDataRepository<ConsumerApps> appsDataRepository, I
...
+        /// <param name="clientId">clientId of App</param>
+        /// <returns>App details as object of ConsumerApp</returns>
+        private async Task<ConsumerApps> GetConsumerByClientIdOfIdentityServerClient(string clientId)
+        {
+            ConsumerApps app = new ConsumerApps();
+            var client = await _clientDataRepository.FirstOrDefaultAsync(x => x.ClientId == clientId);
+            if (client != null)
+            {
+                var scopesAllowed = await _scopes.FetchAsync(x => x.Client.ClientId == clientId);
+                List<AllowedScope> allowedScope = new List<AllowedScope>();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ConsumerAppRepository/ConsumerAppRepository.cs]: unused variable created

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ConsumerAppRepository/ConsumerAppRepository.cs [hunk 4]
```diff
@@ -68,69 +79,210 @@ public ConsumerAppRepository(IDataRepository<ConsumerApps> appsDataRepository, I
...
+        /// <returns>list of string scope</returns>
+        private List<string> ReturnListOfScopesInStringFromEnumAllowedScope(List<AllowedScope> scopes)
+        {
+            List<string> allowedScope = new List<string>();
+            foreach (var item in scopes)
             {
-                _appsDataRepository.UpdateAsync(consumerApps);
-                await _appsDataRepository.SaveChangesAsync();
-                return consumerApps.Id;
+                var value = item.ToString();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ConsumerAppRepository/ConsumerAppRepository.cs]: Unused variable

**Pull request merged**